File: algo-99-bottles/99_bottles.js
function formatBottleString(n) {  
  if (n === 1) {
    return "bottle"
  } 
  return "bottles"
}

function formatCount(n) {
  if (n === 0) {
    return "no more"
  } 
  return n
}

function bottleSong() {
  for (let i = 99; i >= 1; i--) {
    let currentCountBottles = formatBottleString(i)
    let currentCount = formatCount(i)
    console.log(`${currentCount} ${currentCountBottles} of beer on the wall, ${currentCount} ${currentCountBottles} of beer.\nTake one down and pass it around, ${formatCount(i-1)} ${formatBottleString(i-1)} of beer on the wall.`)
  }
  console.log("No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.")
};

bottleSong();

File: algo-99-bottles/99_bottles.py
def bottle_song(n):
	# write your code here!
	if (n == 1):
		objects = 'bottle'
		objectsMinusOne = 'bottles'
	elif (n == 2):
		objects = 'bottles'
		objectsMinusOne = 'bottle'
	else:
		objects = 'bottles'
		objectsMinusOne = 'bottles'

	if (n > 0):
		print(f"{str(n)} {objects} of beer on the wall, {str(n)} {objects} of beer.")
		print(
		    f"Take one down and pass it around, {str(n-1)} {objectsMinusOne} of beer on the wall."
		)
	elif (n == 0):
		print("No more bottles of beer on the wall, no more bottles of beer.")
		print("Go to the store and buy some more, 99 bottles of beer on the wall.")
	else:
		print("Error: Wheres the booze?")

for bottles in range(99, -1, -1):
	bottle_song(bottles)
